File: LeetCode/Easy/938. Range Sum of BST.php
<?php



class Solution
{
    function __construct()
    {
        $this->ans = 0;
    }



    function rangeSumBST( $root, $low, $high )
    {
        if ( !isset( $root ) )
        {
            return $this->ans;
        }

        if ( $root->val > $low )
        {
            $this->rangeSumBST( $root->left, $low, $high );
        }

        if ( $root->val < $high )
        {
            $this->rangeSumBST( $root->right, $low, $high );
        }

        if ( $root->val >= $low && $root->val <= $high )
        {
            $this->ans += $root->val;
        }

        return $this->ans;
    }
}



?>

File: LeetCode/Easy/938. Range Sum of BST.java
class Solution
{
    int ans;

    Solution ()
    {
        this.ans = 0;
    }



    public int rangeSumBST( TreeNode root, int low, int high )
    {
        if ( root == null )
        {
            return this.ans;
        }

        if ( root.val > low )
        {
            this.rangeSumBST( root.left, low, high );
        }

        if( root.val < high )
        {
            this.rangeSumBST( root.right, low, high );
        }

        if( root.val >= low && root.val <= high )
        {
            this.ans += root.val;
        }
        
        return this.ans;
    }
}
